src/util-ebpf.c

```diff
@@ -0,0 +1,733 @@
+            flowstats->count++;
+            flowstats->packets += pkts_cnt;
+            flowstats->bytes += bytes_cnt;
+            SC_ATOMIC_ADD(dev->bypassed, pkts_cnt);
+            found = 1;
+            EBPFDeleteKey(mapfd, &key);
+        }
+        key = next_key;
+    }
+
+    struct bpf_maps_info *bpfdata = LiveDevGetStorageById(dev, g_livedev_storage_id);
+    if (bpfdata) {
+        SC_ATOMIC_SET(bpfdata->ipv4_hash_count, hash_cnt);
+    }
+
+    return found;
+}
+
+/**
+ * Bypassed flows cleaning for IPv6
+ *
+ * This function iterates on all the flows of the IPv4 table
+ * looking for timeouted flow to delete from the flow table.
+ */
+static int EBPFForEachFlowV6Table(LiveDevice *dev, const char *name,
+                                  struct flows_stats *flowstats,
+                                  struct timespec *ctime)
+{
+    int mapfd = EBPFGetMapFDByName(dev->dev, name);
+    struct flowv6_keys key = {}, next_key;
+    int found = 0;
+    unsigned int i;
+    unsigned int nr_cpus = UtilCpuGetNumProcessorsConfigured();
+    if (nr_cpus == 0) {
+        SCLogWarning(SC_ERR_INVALID_VALUE, "Unable to get CPU count");
+        return 0;
+    }
+
+    uint64_t hash_cnt = 0;
+    while (bpf_map_get_next_key(mapfd, &key, &next_key) == 0) {
+        bool purge = true;
+        uint64_t pkts_cnt = 0;
+        uint64_t bytes_cnt = 0;
+        hash_cnt++;
+        struct pair values_array[nr_cpus];
+        memset(values_array, 0, sizeof(values_array));
+        int res = bpf_map_lookup_elem(mapfd, &key, values_array);
+        if (res < 0) {
+            SCLogDebug("no entry in v6 table for %d -> %d", key.port16[0], key.port16[1]);
+            key = next_key;
+            continue;
+        }
+        for (i = 0; i < nr_cpus; i++) {
+            int ret = EBPFBypassedFlowV6Timeout(mapfd, &key, &values_array[i], ctime);
+            if (ret) {
+                pkts_cnt += values_array[i].packets;
+                bytes_cnt += values_array[i].bytes;
+            } else {
+                purge = false;
+                break;
+            }
+        }
+        if (purge) {
+            flowstats->count++;
+            flowstats->packets += pkts_cnt;
+            flowstats->bytes += bytes_cnt;
+            SC_ATOMIC_ADD(dev->bypassed, pkts_cnt);
+            found = 1;
+            EBPFDeleteKey(mapfd, &key);
+        }
+        key = next_key;
+    }
+
+    struct bpf_maps_info *bpfdata = LiveDevGetStorageById(dev, g_livedev_storage_id);
+    if (bpfdata) {
+        SC_ATOMIC_SET(bpfdata->ipv6_hash_count, hash_cnt);
+    }
+    return found;
+}
+
+/**
+ * Flow timeout checking function
+ *
+ * This function is called by the Flow bypass manager to trigger removal
+ * of entries in the kernel/userspace flow table if needed.
+ *
+ */
+int EBPFCheckBypassedFlowTimeout(struct flows_stats *bypassstats,
+                                        struct timespec *curtime)
+{
+    struct flows_stats l_bypassstats = { 0, 0, 0};
```

Comment:
l as a leading variable character is evil

Reply:
With a l like leblond :P